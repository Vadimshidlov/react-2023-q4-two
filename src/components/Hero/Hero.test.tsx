// // import { userEvent } from '@testing-library/user-event';
// import * as AppContext from '@/context-store.tsx';
// import { render } from '@/components/rtl-utils.tsx';
// import {
//   mockContextValue,
//   mockFirstItemContextValue,
// } from '@/components/SearchItems/SearchItems.test.tsx';
// import Hero from '@/components/Hero/Hero.tsx';
// import * as getHeroNumberObject from '@/shared/utils/getHeroNumber.ts';
// import App from '@/App.tsx';

// describe('Hero component tests', () => {
//   afterEach(() => {
//     jest.restoreAllMocks();
//   });

//   test('Should render correct data', () => {
//     const contextValue = {
//       contextData: mockFirstItemContextValue,
//       setContextData: () => {},
//     };

//     const mockGetHeroNumber = jest.spyOn(getHeroNumberObject, 'default');
//     mockGetHeroNumber.mockImplementation(() => 1);

//     jest.spyOn(AppContext, 'useContextData').mockImplementation(() => contextValue);

//     const { getByText } = render(
//       <Hero heroData={mockFirstItemContextValue.searchData[0]} setHeroNumber={() => {}} />,
//       {
//         value: contextValue,
//       }
//     );

//     const heroesName = getByText('Luke Skywalker');
//     expect(heroesName).toBeInTheDocument();

//     const heroesBirthdayDate = getByText('Birthday date: 19BBY');
//     expect(heroesBirthdayDate).toBeInTheDocument();

//     const heroesGender = getByText('Gender: male');
//     expect(heroesGender).toBeInTheDocument();
//   });

//   it('should open details component after click', () => {
//     jest.mock('@/hooks/useFetching.ts', () => ({
//       getData: jest.fn(),
//       totalPages: 10,
//       pagesArray: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
//       searchFormHandler: jest.fn(),
//       fetchError: jest.fn(),
//       setFetchError: jest.fn(),
//       currentPage: 1,
//       setCurrentPage: jest.fn(),
//     }));

//     const contextValue = {
//       contextData: mockContextValue,
//       setContextData: () => {},
//     };

//     // jest.spyOn(AppContext, 'useContextData').mockImplementation(() => contextValue);

//     const { asFragment } = render(<App />, {
//       value: contextValue,
//     });

//     expect(asFragment()).toMatchSnapshot();
//   });
// });
