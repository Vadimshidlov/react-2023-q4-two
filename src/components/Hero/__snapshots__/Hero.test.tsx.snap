// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero component tests should open details component after click 1`] = `
<DocumentFragment>
  <h2
    class="page__title"
  >
    StarWars Heroes
  </h2>
  <div
    class="main-page__container"
  >
    <button
      class="error-button"
      type="button"
    >
      Error
    </button>
    <div
      class="search__container"
    >
      <form
        action=""
        class="search__form"
      >
        <div
          class="search-input__container"
        >
          <svg
            fill="none"
            height="25"
            viewBox="0 0 24 25"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M16.5483 9.80256C16.5483 13.5677 13.5072 16.6051 9.77417 16.6051C6.0411 16.6051 3 13.5677 3 9.80256C3 6.0374 6.0411 3 9.77417 3C13.5072 3 16.5483 6.0374 16.5483 9.80256ZM15.0139 18.079C13.4996 19.0453 11.702 19.6051 9.77417 19.6051C4.37604 19.6051 0 15.2164 0 9.80256C0 4.38876 4.37604 0 9.77417 0C15.1723 0 19.5483 4.38876 19.5483 9.80256C19.5483 12.1971 18.6922 14.3912 17.2702 16.0936L23.4844 22.3511L21.3704 24.4797L15.0139 18.079Z"
              fill="black"
              fill-rule="evenodd"
            />
          </svg>
          <input
            type="text"
            value=""
          />
        </div>
        <button
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="searchItems__container"
    >
      <ul
        class="searchItems__heroes"
      >
        <div
          class="not-found__container"
        >
          <h2
            class="not-found__title"
          >
            Sad news... Heroes are not found
          </h2>
        </div>
      </ul>
    </div>
    <div
      class="page__container"
    />
  </div>
</DocumentFragment>
`;
